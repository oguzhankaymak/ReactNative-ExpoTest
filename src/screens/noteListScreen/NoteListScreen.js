import React, { useState } from 'react';
import { View, Text } from 'react-native';

import AddItem from '../../components/item/AddItem';
import ListItem from '../../components/item/ListItem';
import Layout from '../../components/layout/Layout';
import styles from './styles/NoteListScreenStyle';

const NoteListScreen = () => {
  const [data, setdata] = useState([]);
  return (
    <Layout>
      <View style={styles.container}>
        <AddItem addItem={(newData) => setdata([...data, newData])} />
        <ListItem data={data} removeData={(id) => setdata(data.filter((filterValue) => filterValue.id !== id))} />
      </View>
    </Layout>
  );
};

export default NoteListScreen;
