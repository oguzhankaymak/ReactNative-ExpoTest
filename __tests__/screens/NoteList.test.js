import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';

import NoteListScreen from '../../src/screens/noteListScreen/NoteListScreen';

test('render Note List Screen', () => {
  const { getByTestId, getByText, getAllByTestId, queryByText, toJSON } = render(<NoteListScreen />);

  const input = getByTestId('add-input');
  const button = getByTestId('add-button');

  fireEvent.changeText(input, 'item0');
  fireEvent.press(button);
  const item0 = getByText('item0');
  expect(item0).toBeDefined();

  fireEvent.changeText(input, 'item1');
  fireEvent.press(button);
  const item1 = getByText('item1');
  expect(item1).toBeDefined();

  expect(toJSON).toMatchSnapshot();
  fireEvent.press(getAllByTestId('delete')[0]);
  expect(queryByText('item0')).toBeNull();

  const list = getByTestId('list');
  expect(list.children.length).toBe(1);
});
