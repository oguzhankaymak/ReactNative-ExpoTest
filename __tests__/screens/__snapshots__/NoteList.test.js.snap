// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Note List Screen 1`] = `[Function]`;
